### ISSUE TITLE
Tracking issue: removing ftdetect plugins

### ISSUE THREAD

**Issue by theHamsta** [nvim-treesitter/nvim-treesitter#2259]
Ideally, this plugin would concentrate on its core competence of providing parser meta data and queries. ftdetects should be handled by other plugins, user configuration or in the ideal case in upstream Neovim/Vim https://github.com/nvim-treesitter/nvim-treesitter/pull/2234 , following https://github.com/nvim-treesitter/nvim-treesitter/pull/2234#issuecomment-1010319661

Right now we have the following ftdetect plugins
```
cooklang.vim
fusion.vim
gdresource.vi
gdscript.vim
glimmer.vim
glsl.vim
gowork.vim
graphql.vim
hack.vim
hcl.vim
heex.vim
hjson.vim
json5.vim
ledger.vim
nix.vim
prisma.vim
pug.vim
ql.vim
query.vim
surface.vim
teal.vim
tlaplus.vim
yang.vim
```

After #2234 being merged, we would be left with a few ftdetect plugins for small languages. All other ftdect plugins are already in upstream Neovim. This issue tracks the progress whether those languages can be removed because they've been integrated into upstream Neovim/Vim ftdetect

- [ ] rescript (https://github.com/nvim-treesitter/nvim-treesitter/pull/2170)
- [ ] re2c (https://github.com/nvim-treesitter/nvim-treesitter/pull/2159)
- [ ] .Brewfile (ruby) (#2179)
- [x] rasi (https://github.com/nvim-treesitter/nvim-treesitter/pull/2268)
- [ ] hack
- [x] fish
- [x] fusion
- [ ] gdresource
- [ ] gdscript
- [ ] glimmer
- [ ] glsl
- [ ] gowork
- [ ] graphql
- [ ] hcl
- [ ] heex
- [ ] hjson
- [x] #2342 (added to Neovim https://github.com/neovim/neovim/pull/17180 for 0.7.0)
- [ ] ledger
- [ ] nix
- [ ] prisma
- [ ] pug
- [ ] ql
- [x] query (https://github.com/neovim/neovim/pull/17905)
- [ ] surface
- [ ] teal
- [ ] tlaplus
- [ ] yang
- [ ] hocon
- [x] eex (https://github.com/nvim-treesitter/nvim-treesitter/pull/2351)

Contributors can easily create PRs like https://github.com/vim/vim/pull/9501 to make Vim (and then Neovim, which pulls in these updates regularly) support those file types natively.

**Comment by theHamsta** [comment]
I couldn't find `nix` files in Neovim which I couldn't believe since nix is really wide spread. Maybe I just oversaw it.

**Comment by clason** [comment]
If it's not in Vim, it's not in Neovim ;) Currently, filetype detection is taken fully from upstream, with no additions. That might change once we fully switch to `filetype.lua` (which probably won't happen for 0.7).

**Comment by clason** [comment]
Adding to that: there's a `vim-nix` plugin that people use; there is precedent of including (the basic features of) such plugins upstream, but someone has to have the motivation to do so. (For example, I did it for Julia because I didn't like the way that plugin forced LaTeX->Unicode expansion.)

**Comment by clason** [comment]
One thing we could do is to add the detection to `filetype.lua` now and require nvim-treesitter users to opt-in to that if they need the detection? (If you use one experimental feature, you might as well use two...)

@gpanders

**Comment by theHamsta** [comment]
https://github.com/neovim/neovim/pull/17182 seems to add the filetypes in this list.

**Comment by dundargoc** [comment]
> https://github.com/neovim/neovim/pull/17182 seems to add the filetypes in this list. Thank you a lot @dundargoc!

No problem. Just a heads up: there are two filetypes still missing. Mentioned them in the json issue/pr (can't link them at the moment but should be easy to find.)

**Comment by clason** [comment]
I suggest pulling the plug after Neovim 0.7 is released, which ships with `vim.filetype.add()` that makes it easy to add filetypes not recognized by default.

**Comment by clason** [comment]
...which raises a point: what do we want to do with `query` filetype? Should we turn this into a proper filetype plugin support (e.g, upstreaming the `omnifunc` from playground)? 

Would it make sense to even upstream this to Neovim?

**Comment by clason** [comment]
Too late: https://github.com/neovim/neovim/pull/17905 :)

**Comment by clason** [comment]
I believe all of the listed filetypes (including cooklang!) are now supported out of the box by the current HEAD of Neovim 0.7, which will be released on Friday.

I therefore propose we remove all filetype detection (and raise the minimal required Neovim version) on **April 30** (together with the removal of `ensure_installed = maintained`).

**Comment by theHamsta** [comment]
Yes, we can now merge all things that we kept for compatibility (e.g. `vim.ui` functions, `get_note_text` etc). We can also make use of subcapture fallbacks `@type.super.nested.specialization`

**Comment by clason** [comment]
Closing this in favor of #2793; I might pull the plug on this sooner than April 30 as this is not hard to work around even on 0.6.0.
